src/main/webapp/app/exercises/shared/participation/participation.component.ts

```diff
@@ -100,6 +103,24 @@ export class ParticipationComponent implements OnInit, OnDestroy {
                 this.exercise = exerciseResponse.body!;
                 this.participationService.findAllParticipationsByExercise(params['exerciseId'], true).subscribe((participationsResponse) => {
                     this.participations = participationsResponse.body!;
+                    if (this.exercise.type === ExerciseType.PROGRAMMING) {
+                        const programmingExercise = this.exercise as ProgrammingExercise;
+                        if (programmingExercise.projectKey) {
+                            this.profileService.getProfileInfo().subscribe((profileInfo) => {
+                                for (let i = 0; i < this.participations.length; i++) {
+                                    const programmingParticipation = this.participations[i] as ProgrammingExerciseStudentParticipation;
+                                    if (programmingParticipation.buildPlanId) {
+                                        programmingParticipation.buildPlanUrl = createBuildPlanUrl(
+                                            profileInfo.buildPlanURLTemplate,
+                                            programmingExercise.projectKey!,
+                                            programmingParticipation.buildPlanId,
+                                        );
+                                        this.participations[i] = programmingParticipation;
+                                    }
+                                }
+                            });
+                        }
+                    }
```

Comment:
This logic seems to be duplicated with the one here: https://github.com/ls1intum/Artemis/pull/3914/files#diff-3d3907f08211c14b03878ac26c33f79e894395584100438962b80e1e5190fcbaR107-R121

Do you think we can extract it in order to avoid duplication?

Reply:
@Olimbia23 @MaisiKoleni What do you think about this implementation? I wanted to extract more but we iterate there over different arrays (results and participations) so unless you have a practical idea I think that's all I can do here.

Reply:
Since the objects are mutable, we could map the results to their corresponding participations and pass that to a general method that works with the participation array. Then we could extract even more common code.

Reply:
Yes, I thought about that. The only issue I have is this being addition calculations on the client. But I can take a look, maybe I can refactor the component to use participation as the other ones so this issue doesn't arise.

Reply:
done